express-post-json/index.js

```diff
@@ -0,0 +1,27 @@
+const express = require('express');
+const app = express();
+
+let nextId = 1;
+const grades = {};
+
+app.get('/api/grades', (req, res) => {
+  const gradesArray = [];
+  for (let i = 0; i < nextId; i++) {
+    gradesArray.push(grades);
+  }
+  res.json(grades);
+});
+
+app.use(express.json());
+
+app.post('/api/grades', (req, res) => {
+  grades[nextId] = req.body;
+  req.body.id = nextId++;
+  res.json(req.body);
+  res.send().status(201);
```

Comment:
Be sure to set the status code before `res.json()`. The response is sent from top to bottom and the status code goes on the start line, so we need to specify the status code before the body.

Reply:
Ah got it. Thanks Tim